client/state/ui/gutenberg-blocks-warning-dialog/actions.js

```diff
@@ -0,0 +1,20 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import { GUTENBERG_BLOCKS_WARNING_DIALOG_IS_SHOWING } from 'state/action-types';
+
+export function showGutenbergBlocksWarningDialog() {
+	return {
+		type: GUTENBERG_BLOCKS_WARNING_DIALOG_IS_SHOWING,
```

Comment:
👍 if we'd like the editor to toggle display of this. I'm pretty open to options here, for example making the dialog self-contained is also okay (just use component state). The latter would require us to have the dialog responsible for either using passed in post_content / fetching itself + using: 

`import { parse } from '@wordpress/blocks';` or a lighter weight check of `!! content && content.indexOf( '<!-- wp:' ) !== -1;`

Feel free to pick whatever feels better.

Reply:
Thanks for the suggestion!

I definitely like having components self contained, so they can work out of the box by just importing them (btw, is not that the definition of a component? 😅).

I moved the whole the state to the component itself in 77446ccd1e857353f1aa4a8b107cb4fea570a419